File: VB/E3647/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Diagnostics
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraGrid.Views.Layout.ViewInfo
Imports DevExpress.XtraGrid.Views.Layout.Modes
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraGrid.Views.Grid.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraGrid.Drawing
Imports DevExpress.XtraEditors.Controls

Namespace GridSample

    Public Partial Class Form1
        Inherits Form

        Private helper As ColumnLineHelper

        Public Sub New()
            InitializeComponent()
            gridControl1.DataSource = CreateTable(20)
            helper = New ColumnLineHelper(gridView1)
        End Sub

        Private Shared Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(String))
            tbl.Columns.Add("Number", GetType(String))
            tbl.Columns.Add("Date", GetType(String))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i, i Mod 2, Date.Now.AddDays(i).ToString()})
            Next

            Return tbl
        End Function

        Private Sub gridControl1_Paint(ByVal sender As Object, ByVal e As PaintEventArgs)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridView1.Columns("Number").Group()
            gridView1.ExpandAllGroups()
            For Each column As GridColumn In gridView1.Columns
                checkedListBoxControl1.Items.Add(column, column.ToString(), CheckState.Unchecked, True)
            Next

            checkedListBoxControl1.Items(0).CheckState = CheckState.Checked
        End Sub

        Private columnList As List(Of GridColumn) = New List(Of GridColumn)()

        Private Sub checkedListBoxControl1_ItemCheck(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.ItemCheckEventArgs)
            columnList = New List(Of GridColumn)()
            For Each item As CheckedListBoxItem In checkedListBoxControl1.CheckedItems
                columnList.Add(TryCast(item.Value, GridColumn))
            Next

            helper.Columns = columnList
        End Sub

        Private Sub checkButton1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            helper.Hide = checkButton1.Checked
        End Sub
    End Class
End Namespace

File: CS/E3647/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Data;
using DevExpress.XtraGrid.Views.Grid;
using System.Collections;
using DevExpress.XtraGrid;
using System.Diagnostics;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.Utils.Drawing;
using DevExpress.XtraGrid.Views.Layout.ViewInfo;
using DevExpress.XtraGrid.Views.Layout.Modes;
using System.Threading;
using System.Reflection;
using DevExpress.XtraGrid.Columns;
using DevExpress.Utils;
using DevExpress.XtraEditors;
using System.Runtime.InteropServices;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.XtraEditors.Repository;
using DevExpress.Utils.Menu;
using DevExpress.XtraGrid.Menu;
using System.IO;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraGrid.Views.Grid.Drawing;
using DevExpress.Utils.Paint;
using DevExpress.XtraGrid.Views.Card.ViewInfo;
using System.ComponentModel;
using DevExpress.XtraGrid.Drawing;
using DevExpress.XtraEditors.Controls;

namespace GridSample
{
    public partial class Form1 : Form
    {
        ColumnLineHelper helper;
        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
            helper = new ColumnLineHelper(gridView1);
        }
        private static DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(string));
            tbl.Columns.Add("Number", typeof(string));
            tbl.Columns.Add("Date", typeof(string));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, i % 2, DateTime.Now.AddDays(i).ToString() });
            return tbl;
        }

        private void gridControl1_Paint(object sender, PaintEventArgs e)
        {
        }




        private void Form1_Load(object sender, EventArgs e)
        {
            gridView1.Columns["Number"].Group();
            gridView1.ExpandAllGroups();
            foreach (GridColumn column in gridView1.Columns)
                checkedListBoxControl1.Items.Add(column, column.ToString(), CheckState.Unchecked, true);
            checkedListBoxControl1.Items[0].CheckState = CheckState.Checked;

        }

        List<GridColumn> columnList = new List<GridColumn>();
        private void checkedListBoxControl1_ItemCheck(object sender, DevExpress.XtraEditors.Controls.ItemCheckEventArgs e)
        {
            columnList = new List<GridColumn>();
            foreach (CheckedListBoxItem item in checkedListBoxControl1.CheckedItems)
                columnList.Add(item.Value as GridColumn);
            helper.Columns = columnList;
        }

        private void checkButton1_CheckedChanged(object sender, EventArgs e)
        {
            helper.Hide = checkButton1.Checked;
        }
   }
}
